### ISSUE TITLE
HMW6 

### ISSUE THREAD

**Issue by juansbr7** [juansbr7/STAT547Bonilla_Juan#1]

**Comment by KateJohnson** [comment]
Hi Juan,

Great job, this is a lot of work! Some comments below:

**Part 1:**
Wow this is some serious string work, good job! However, I did find it hard to follow without seeing the output of your code. Do you know why you weren’t able to generate an rmd file? Was it just because the file would be too big?

**Part 2:**
- In terms of dealing with NAs, my process is usually to reduce the dataset to only my variables of interest and then exclude all rows with NA observations. Most functions (ie. linear regression) have na.rm=TRUE automatically. 
- Again, It would have been really nice to see the output of your code. Perhaps you could have shown the output of str(candy). Or done kable(head(candy))
- I’d be a bit worried about your method for changing age from character to integer. I wonder if maybe some entries that really could have been integer (ex., “7 “) got changed to NA as well. 
- I don’t think you age histogram was too wacky, you probably could just specify more bins.
- I like your logic for changing NA to neutral
- I liked your plot, but could you use position=“dodge” in geom_bar? It would be a lot easier to compare the bar heights. 
- I liked your exploration of lots of cool plots, in general I had issue with interpreting the differences in colours or circle area, but it’s probably OK for exploration purposes.

But on the whole, great job! 👍

**Comment by emilymistick** [comment]
hi @juansbr7,

Nice job on this homework! Very thorough work!

**Part 1**
- Important to get that .md problem fixed. Try using github_document as your YAML output at the top of the .Rmd? That's what I've been doing, then using knitr to create the .md file.
- This is a very good and thorough demonstration of string manipulation. You clearly understand how to deal with strings and the `stringr` package!
- Also nice work with regular expressions. I'm still trying to learn those so reading this assignment was helpful to me, thanks.

**Part 2**
- In RMarkdown you can use message=FALSE in the code chunk header rather than the `supressMessages` function, for a shorter output of your libraries
- It would be nice if you described the task before diving right in (like you did in part 1). A high level summary on your README would be a big improvement. and/or at the beginning of your markdown file.
- Good use of `stringr`!
- It would be useful to see some of the data before and after. As Kate suggested, try using `head()` and select variables of interest (`select()`) to make a small example table.
- Good work making an exploratory histogram and filtering age after seeing unusual results.
- Use more `kable()`! It makes the data outputs much nicer and easier to follow. You can use it anytime you're displaying a table.
- I like you analysis of NA -> neutral as an example of modifying a dataframe and including the data in the plot. Although if the person doesn't know the candy it's more than they have no info than that they are neutral?
- For your millenials plot the colors are very hard to interpret. Good practice making this type of graph though for exploratory purposes.
- Color and dot sizes are also *really* hard to make sense of. Points here for trying a new plot, but not for informative design.


Overall, great work! I noticed that both exercises focused on string management. I'd recommend practicing function writing, nested dataframes, and mapping. :-)

**Comment by derekcho** [comment]
Hi @juansbr7 ! Here are some comments about your hw06:

-	Nice work in the first task despite the issues with knitting a .md file. Some other students had similar issues as well. You answered the questions correctly and provided good explanations of the stringr functions as well. Nice work for this part!
-	Your second task (handling the candy data) was very well done! I thought that you were able to explain each step of your process well and you did some pretty cool things with the data. I liked how you used a variety of methods including regex and replacement to clean your dataset. You did a good job of exploring the data as well. You are right about the second plot (for millennials) not being readble, the first plot should be sufficient. Be careful about setting two different aesthetics to a single variable (in your case, n). I think that it is redundant and should not be done unless you have a really good reason to. However, overall this task was very well done!
-	Good work overall! Your assignment was well organized, and you provided decent explanations as well. Although your plots are quite easy to read, I think that they would benefit from having some interpretation of the results as well. However, I could not find your reflections for this assignment.

Note: your mark (check minus, check, check plus) will be distributed later.
